curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-map-method-to-extract-data-from-an-array.english.md

```diff
@@ -24,13 +24,13 @@ The <code>watchList</code> array holds objects with information on several movie
 ```yml
 tests:
   - text: The <code>watchList</code> variable should not change.
-    testString: assert(watchList[0].Title === "Inception" && watchList[4].Director == "James Cameron", 'The <code>watchList</code> variable should not change.');
+    testString: assert(watchList[0].Title === "Inception" && watchList[4].Director == "James Cameron");
   - text: Your code should not use a <code>for</code> loop.
-    testString: assert(!code.match(/for\s*?\(.+?\)/g), 'Your code should not use a <code>for</code> loop.');
+    testString: assert(!removeJSComments(code).match(/for\s*?\(.+?\)/));
```

Comment:
```suggestion
    testString: assert(!removeJSComments(code).match(/for\s*?\(.*?\)/));
```
I guess this is better than what we have, this will match empty `for`s too. Since the test is `Your code should not use a <code>for</code> loop` even an empty for is technically a for

Reply:
@thecodingaviator Fair enough.

Reply:
@thecodingaviator Made the change.